File: lib/mappers/product_mapper.dart
import 'package:emarsys_sdk/model/predict/product.dart';

import 'mapper.dart';

class ProductMapper extends Mapper<List<Product>, List<Map<String, dynamic>>> {
  @override
  List<Product> map(List<dynamic> input) {
    return input
        .where((element) => element != null && (element as Map).isNotEmpty)
        .map((productMap) => Product(
            productId: productMap["productId"],
            title: productMap["title"],
            linkUrl: productMap["linkUrl"],
            feature: productMap["feature"],
            cohort: productMap["cohort"],
            customFields: productMap["customFields"],
            imageUrlString: productMap["imageUrlString"],
            imageUrl: productMap["imageUrl"],
            zoomImageUrlString: productMap["zoomImageUrlString"],
            zoomImageUrl: productMap["zoomImageUrl"],
            categoryPath: productMap["categoryPath"],
            available: productMap["available"],
            productDescription: productMap["productDescription"],
            price: productMap["price"],
            msrp: productMap["msrp"],
            album: productMap["album"],
            actor: productMap["actor"],
            artist: productMap["artist"],
            author: productMap["author"],
            brand: productMap["brand"],
            year: productMap["year"]))
        .toList();
  }
}

File: android/src/main/kotlin/com/emarsys/emarsys_sdk/mapper/ProductMapper.kt
package com.emarsys.emarsys_sdk.mapper

import com.emarsys.predict.api.model.Product

class ProductMapper : Mapper<Product, Map<String, Any?>> {
    override fun map(input: Product): Map<String, Any?> {
        return mutableMapOf(
            "productId" to input.productId,
            "title" to input.title,
            "linkUrlString" to input.linkUrl,
            "feature" to input.feature,
            "cohort" to input.cohort,
            "customFields" to input.customFields,
            "imageUrlString" to input.imageUrl.toString(),
            "zoomImageUrlString" to input.zoomImageUrl.toString(),
            "categoryPath" to input.categoryPath,
            "available" to input.available,
            "productDescription" to input.productDescription,
            "price" to input.price,
            "msrp" to input.msrp,
            "album" to input.album,
            "actor" to input.actor,
            "artist" to input.artist,
            "author" to input.author,
            "brand" to input.brand,
            "year" to input.year
        )
    }
}